crates/ruff_linter/src/rules/flake8_comprehensions/rules/unnecessary_comprehension.rs

```diff
@@ -30,10 +29,30 @@ use crate::rules::flake8_comprehensions::fixes;
 /// set(iterable)
 /// ```
 ///
+/// ## Known problems
+///
+/// This rule may produce false positives for dictionary comprehensions that iterate over a mapping.
+/// The `dict` constructor behaves differently depending on if it receives a sequence (e.g., a
+/// `list`) or a mapping (e.g., a `dict`). When the comprension iterates over the keys of a mapping,
+/// replacing it with a `dict` constructor will give a different result.
+///
+/// For example:
+///
+/// ```pycon
+/// >>> d1 = {(1, 2): 3, (4, 5): 6}
+/// >>> {x: y for x, y in d1}  # Iterates over the keys of a mapping
+/// {1: 2, 4: 5}
+/// >>> dict(d1)               # Suggested fix
```

Comment:
I wrote it this way initially as "Incorrect suggested fix" but I'm not sure it's better.

Reply:
I think otherwise it could possibly be misread by somebody skim-reading as saying that it's what we're suggesting here, in the docs. And that's obviously not the case 😄

Reply:
I pushed a fix that should resolve review comments.